qcfractal/interface/collections/reaction_dataset.py

```diff
@@ -239,16 +372,120 @@ def visualize(self,
 
         return self._visualize(metric, bench, query=query, groupby=groupby, return_figure=return_figure, kind=kind)
 
```

Comment:
Does not return the correct molecule, e.g.

```
s22 = client.get_collection("ReactionDataset", "S22")
mol = s22.get_molecules(subset='Ammonia Dimer', stoich='default')
print(mol, mol.id)
```
gives:
```
    Geometry (in Angstrom), charge = 0.0, multiplicity = 1:

       Center              X                  Y                   Z       
    ------------   -----------------  -----------------  -----------------
    O                -1.397621297827    -1.885836800365    -0.367306098301
    N                -1.464255001147     0.364182799152     0.019230098748
    C                -4.185739797656     0.369666900716     0.036095997764
    C                -3.483259802214     1.578311100122     0.250075200651
    C                -2.117950202456     1.530704799511     0.233838297543
    C                -2.077383297566    -0.863749200485    -0.189941402177
    C                -3.515603197998    -0.805195001276    -0.175758500412
    H                -5.267804501347     0.370742797362     0.041141898264
    H                -3.992033400169     2.512755999947     0.421441397921
    H                -1.492919599297     2.398409599646     0.388501799340
    H                -4.040122601443    -1.734845201024    -0.337926899246
    H                -0.426526600411     0.361212702097     0.007353801068
    N                 1.432761601549     0.363970297460    -0.015950798157
    C                 2.115419999833    -0.780344998216     0.168109900012
    C                 3.523758601997    -0.801609598128     0.154502700718
    C                 4.218589700700     0.373578298193    -0.052592901512
    C                 3.509970799813     1.561501399725    -0.244976297763
    C                 2.128013798309     1.495332398665    -0.217537401035
    H                 4.045920600345    -1.736135599652     0.307688300525
    H                 5.299942597541     0.366600900875    -0.066334898116
    H                 4.011092299506     2.502431297474    -0.413005201910
    H                 1.533987799051     2.389383699136    -0.367056501286
    N                 1.388312298093    -1.908303802027     0.419814897679
    H                 1.869471402447    -2.781277302142     0.294038502466
    H                 0.408906697955    -1.907994201608     0.130085998541
 24
```

This is not caught by the test suite, as there is not test for `ReactionDataset.get_molecules`

Reply:
I think this relates to `subset` not being used in `_molecule_indexer`, see below.